RecFCCeeCalorimeter/src/components/AugmentClustersFCCee.cpp

```diff
@@ -177,48 +360,261 @@ StatusCode AugmentClustersFCCee::execute([[maybe_unused]] const EventContext &ev
+          E_cell_Min[layer+startPositionToFill] = 0.;
+        } else {
+          std::vector<double> sortedVec = local_E_Max[layer+startPositionToFill];
+          // move the top 2 max to the beginning
+          std::partial_sort(sortedVec.begin(), sortedVec.begin()+2, sortedVec.end(), std::greater<double>());
+          E_cell_Max[layer+startPositionToFill] = sortedVec[0];
+          E_cell_secMax[layer+startPositionToFill] = sortedVec[1];
+          // get the corresponding theta IDs
+          auto it_Max   = std::find(local_E_Max[layer+startPositionToFill].begin(), local_E_Max[layer+startPositionToFill].end(), sortedVec[0]);
+          int index_Max = std::distance(local_E_Max[layer+startPositionToFill].begin(), it_Max);
+          auto it_secMax   = std::find(local_E_Max[layer+startPositionToFill].begin(), local_E_Max[layer+startPositionToFill].end(), sortedVec[1]);
+          int index_secMax = std::distance(local_E_Max[layer+startPositionToFill].begin(), it_secMax);
+          E_cell_Max_theta[layer+startPositionToFill]    = local_E_Max_theta[layer+startPositionToFill][index_Max];
+          E_cell_secMax_theta[layer+startPositionToFill] = local_E_Max_theta[layer+startPositionToFill][index_secMax];
+          // find the E_min inside the theta range of E_cell_Max and E_cell_secMax
+          // the theta_E_pair are sorted in ascending order of thetaID
+          for (size_t i = 0; i < theta_E_pair[layer+startPositionToFill].second.size(); i ++) {
+            if (
+              (theta_E_pair[layer+startPositionToFill].first[i] > std::min(E_cell_Max_theta[layer+startPositionToFill], E_cell_secMax_theta[layer+startPositionToFill])) &&
+              (theta_E_pair[layer+startPositionToFill].first[i] < std::max(E_cell_Max_theta[layer+startPositionToFill], E_cell_secMax_theta[layer+startPositionToFill])) &&
+              (theta_E_pair[layer+startPositionToFill].second[i] < E_cell_Min[layer+startPositionToFill])
+            )
+            {
+              E_cell_Min[layer+startPositionToFill] = theta_E_pair[layer+startPositionToFill].second[i];
+            }
+          }
+        }
+        if (E_cell_Min[layer+startPositionToFill] > 1e12)    E_cell_Min[layer+startPositionToFill] = 0.;  // check E_cell_Min
+      }  // end of loop over layers
+    }
+
+    // local maxima of E vs phi (could be more than one) and the corresponding module
+    // note that code should be modified to handle periodicity of module/phi (and tested on particles in the direction of the 1536..0 module transition)
+    std::vector<std::pair<std::vector<int>, std::vector<double>>> module_E_pair;
+    std::vector<std::vector<double>> local_E_Max_vs_phi(numLayersTotal, std::vector<double>());
+    std::vector<std::vector<int>> local_E_Max_vs_phi_module(numLayersTotal, std::vector<int>());
+
+    startPositionToFill = 0;
+    for (size_t k = 0; k < m_readoutNames.size(); k++) {
```

Comment:
this for instance looks like a block of code that should be only executed if do_pi0_photon_shapeVar is true, otherwise the input vectors will be empty. This might explain the crashes (ReferenceError: attempt to access a null-pointer) observed when the algorithm is configured with do_pi0_photon_shapeVar = false